### DISCUSSION TITLE
Obs_Quality flags

### DISCUSSION THREAD

**Discussion by robdarvell** [dtcenter/METplus#1005] [Configuration] [answered]
Reading through the documentation, it appears as though you can specify specific obs quality flags to be included within the verification.
Is there any specific way at present in order to not include specific quality flags. With our data we can potentially have any number of specific flags for specifying this element (far too many to list), but we do know that any number which ends up as an odd number has been quality controlled out (rejected).
Within our verification at present we can include or remove these obs from the stats, but was wondering if a similar sort of feature would be worth adding if not currently available. 
I am not suggesting anything any more complex than adding some sort of _not_ key which the code then picks up and uses: eg: !"99" so you wouldn't include values of 99 but include everything else.

**Comment by j-opatz** [comment]
Hi Rob, and thanks for your question. 

I'm going to include @JohnHalleyGotway on this discussion as I'm not sure of a quick way to do that without resorting to some kind of Python pre-processing. I'm assuming this is for a Point2Grid usage, where the -qc flag can be utilized?

**Comment by JohnHalleyGotway** [comment, marked as answer]
@robdarvell John Opatz is right. That functionality, to provide a list of quality control flag values to exclude, is NOT currently supported. Instead, MET version 10.0.0 expects you to specify the ones that should be included. Tara brought up this question at a project meeting last week, and I recommended that we add a new configuration option to handle it. This is really similar to configuring the station id's to be included (sid_inc) versus those to be excluded (sid_exc) in Point-Stat.

Following that paradigm, we could rename the existing "obs_quality" as "obs_quality_inc", and add a new config option for "obs_quality_exc". Even though we would renamed "obs_quality", we should still parse it and store its value in "obs_quality_inc" for backwards compatibility.

I wrote up this feature request in a new GitHub issue:
https://github.com/dtcenter/MET/issues/1858

Please read through that issue description and feel free to add a comment there with any additional information, questions, or concerns.
